pkg/trace/obfuscate/sql_tokenizer.go

```diff
@@ -327,8 +327,12 @@ func (tkn *SQLTokenizer) scanLiteralIdentifier(quote rune) (TokenKind, []byte) {
 	return ID, t
 }
 
+func isValid(r rune) bool {
+	return r != EOFChar && r != utf8.RuneError
```

Comment:
`r == utf8.RuneError` does not mean that the character is not a valid rune. `utf8.RuneError` is equal to � (replacement char) which _is_ valid UTF8. The only way to find out if a rune is valid is by using [utf8.ValidRune](https://golang.org/pkg/unicode/utf8/#ValidRune). That being said, it should not be necessary to do that. The key is in the `advance` function where you use [utf8.DecodeRune](https://golang.org/pkg/unicode/utf8/#DecodeRune): the UTF8 character being read is only invalid if the return is `utf8.RuneError` _AND_ `n == 0` or `n == 1`

Lastly, I'd add an option in using the replacement char to protect from a mistake like this in the future. As a side note, even the obfuscator will insert the replacement char and easily lead to this scenario.

https://play.golang.org/p/dwsCraxeiKm

Reply:
Ahh, I misunderstood the purpose of `utf8.RuneError`. I've fixed that in the code and also added tests to cover the replacement char case.